sktime/forecasting/base/adapters/_generalised_statsforecast.py

```diff
@@ -254,3 +271,197 @@ def _predict_interval(self, fh, X, coverage):
         final_interval_predictions = pandas.concat(interval_predictions, copy=False)
 
         return final_interval_predictions
+
+
+class StatsForecastBackAdapter(_TS):
+    """StatsForecast Back Adapter.
+
+    StatsForecastBackAdapter is a wrapper for sktime forecasters to be used in
+    StatsForecast composite models.
+
+    Parameters
+    ----------
+    estimator : sktime estimator
+
+    Examples
+    --------
+    >>> from sktime.datasets import load_airline
+    >>> from sktime.forecasting.statsforecast import StatsForecastMSTL
+    >>> from sktime.forecasting.ets import AutoETS
+
+    >>> y = load_airline()
+    >>> trend_forecaster = AutoETS() # doctest: +SKIP
+    >>> model = StatsForecastMSTL( # doctest: +SKIP
+            season_length=[3,12],
+            trend_forecaster=trend_forecaster
+        )
+    >>> fitted_model = model.fit(y=y) # doctest: +SKIP
+    >>> y_pred = fitted_model.predict(fh=[1,2,3]) # doctest: +SKIP
+    """
+
+    _tags = {
+        "python_dependencies": ["statsforecast", "statsmodels"],
+    }
+
+    def __init__(self, estimator):
+        self.estimator = estimator
+        self.in_sample_fh = None
+        self.in_sample_X = None
+
+        super().__init__()
+
+    def __repr__(self):
+        return "StatsForecastBackAdapter"
+
+    def fit(self, y, X=None):
+        """Fit to training data.
+
+        Parameters
+        ----------
+        y : ndarray
+            Clean time series of shape (t, ).
+        X : typing.Optional[numpy.ndarray], default=None
+
+        Returns
+        -------
+        self : returns an instance of self.
+        """
+        from sklearn.base import clone
+
+        self.in_sample_fh = np.flip(-np.arange(len(y)))
+        estimator = clone(self.estimator)
+        clone_self = self.new()
+        clone_self.estimator = estimator.fit(y=y, X=X)
```

Comment:
Clarifying a doubt: at this point, `self.estimator` is not trained then? Only, `clone_self.estimator` is fitted? Is that correct understanding?

Reply:
Yes, I did this so as to not overwrite `estimator`, a parameter of the adapter. However, if those tests (checking if parameters are overwritten) are not run on the adapter because it doesn't inherit from any base sktime class, I agree that `self.estimator` should be trained. I'll try and see if training `self.estimator` works.